pkg/stacker/publisher.go

```diff
@@ -75,255 +66,6 @@ func NewPublisher(opts *PublishArgs) *Publisher {
 	}
 }
 
-func basicAuth(username, password string) string {
-	auth := username + ":" + password
-	return base64.StdEncoding.EncodeToString([]byte(auth))
-}
-
-func clientRequest(method, url, username, password string, headers map[string]string, body io.Reader) (*http.Response, error) {
-	req, err := http.NewRequestWithContext(context.TODO(), method, url, body)
-	if err != nil {
-		log.Errorf("unable to create http request err:%s", err)
-		return nil, err
-	}
-
-	// FIXME: handle bearer auth also
-	if username != "" && password != "" {
-		req.Header.Add("Authorization", "Basic "+basicAuth(username, password))
-	}
-
-	if len(headers) > 0 {
-		for k, v := range headers {
-			req.Header.Add(k, v)
-		}
-	}
-
-	res, err := http.DefaultClient.Do(req)
-	if err != nil {
-		log.Errorf("http request failed url:%s", url)
-		return nil, err
-	}
-
-	return res, nil
-}
-
-func fileDigest(path string) (*godigest.Digest, error) {
```

Comment:
its strange to move this function 'fileDigest' from 'publisher.go' to 'referrer.go'.
Maybe you felt that it didn't really belong in "publisher.go", but to move it to another file with another name that doesn't really fit is strange.

Either leave it here (possibly to be cleaned up later) or move them to a better named file.

this could also be argued for clientRequest and publishArtifact  and 'basicAuth'

Reply:
Publishing involves two separate steps now.
Publish the image/tag with libcontainers APIs (old docker libs) and the new referrers where we have to handcraft HTTP requests. This has been moved into referrers.go now. fileDigest() was a helper routine for the second path.